### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by MarcoOquendoC**
# [Blog app - integration specs for views and fix n+1 problems](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md)

## Hi good reviewer,

P and Bob thinking

Pair programming with @Miliyonayalew.
## This Pull Request proposes the following changes:

- Make sure that the n+1 queries problem is solved when fetching all posts and their comments for a user.
- Secure app from n+1 queries problems in the index and show pages of Post.
- Write integration tests using Capybara gem.
- Avoid usage of bullet gem in this exercise - analyze the logs instead.
- Include integration specs for other features such as Pagination

### CODE REVIEW

File: spec/views/users/show.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe 'User index page', type: :feature do
```
Comment (Deepakdanger) [inline spec/views/users/show.html.erb_spec.rb]: -  [x] There is a typo here. Instead of "User index page', it should be 'User show page'.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @MarcoOquendoC, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️

- [x] There should be the integration spec for post index and show page as well. Please write the test for that as well.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/users/show.html.erb_spec.rb (hunk 1 above)
Comment (MarcoOquendoC) [reply spec/views/users/show.html.erb_spec.rb]: Done

**Review by mugishasam123 — APPROVED** [review]
Hi @MarcoOquendoC  :wave: , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors :heavy_check_mark: 
- Descriptive PR summary , keep up the energy :rocket: 

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**